### PR TITLE
added support for more component types

### PR DESCRIPTION
**Pull request by mizrael**

### CODE REVIEW

**Review by sagar-bakliwal — APPROVED** [review]

File: src/Persistence/Templates/BuiltInTemplateIds.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.PowerPlatform.PowerApps.Persistence.Templates;
+
+public sealed class BuiltInTemplateIds
```
Comment (tehcrashxor) [inline src/Persistence/Templates/BuiltInTemplateIds.cs:6]: `class` for holding `const`s and/or `static`s shouldn't be instantiable: 
```suggestion
public static class BuiltInTemplateIds
```

File: src/Persistence/Templates/BuiltInTemplateNames.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.PowerPlatform.PowerApps.Persistence.Templates;
+
+public sealed class BuiltInTemplateNames
```
Comment (tehcrashxor) [inline src/Persistence/Templates/BuiltInTemplateNames.cs:6]: `class` for holding `const`s and/or `static`s shouldn't be instantiable: 
```suggestion
public static class BuiltInTemplateNames
```

File: src/Persistence/Templates/BuiltInTemplates.cs [hunk 3]
```diff
@@ -5,18 +5,18 @@ namespace Microsoft.PowerPlatform.PowerApps.Persistence.Templates;
 
 public class BuiltInTemplates
```
Comment (tehcrashxor) [inline src/Persistence/Templates/BuiltInTemplates.cs:6]: `class` for holding `const`s and/or `static`s shouldn't be instantiable: 
```suggestion
public static class BuiltInTemplates
```

**Review by tehcrashxor — APPROVED** [review]

File: src/Persistence/Templates/BuiltInTemplateIds.cs (hunk 1 above)
Comment (mizrael) [reply src/Persistence/Templates/BuiltInTemplateIds.cs:6]: good point

**Review by tehcrashxor — APPROVED** [review]

**Review by sagar-bakliwal — APPROVED** [review]

**Review by swapnilbhalgat — APPROVED** [review]

**Review by joem-msft — APPROVED** [review]

**Pull request merged**